agreement/player.go

```diff
@@ -40,7 +49,7 @@ type player struct {
 
 	// Deadline contains the time of the next timeout expected by the player
 	// state machine (relevant to the start of the current period).
-	Deadline time.Duration
+	Deadline Deadline
```

Comment:
Just to clarify in response to Pavel's comment:

> checked, if restore fails, it does not abort the node, only logs.

One worry is that, even though the node continues running, it has forgotten its state (e.g., what proposals it voted for).  So, if a node software upgrade (which involves killing and restarting algod) happens in the middle of a round (which is fairly likely; we're often in the middle of some round), then the restarted algod won't know what it voted for before it was killed, and could sign a conflicting message.  In our protocol, a pair of conflicting signatures in the same round/period/step counts as a "equivocation" and can be used by anyone to treat this node's signature as a signature on any message of their choice (by the reason that this node seems to be corrupted and is signing conflicting messages already).  If this happens to many nodes in the same round (e.g., due to a synchronized restart of many nodes at the same time), this might allow an adversary to construct seemingly-valid certificates without having to compromise a large fraction of voting nodes, etc.

Reply:
merged the persistence change from PR #5669 cce/cce/taggedDeadlines-twofields into this branch to address this issue